src/classlibnative/bcltype/number.cpp

```diff
@@ -304,13 +304,20 @@ void DoubleToNumber(double value, int precision, NUMBER* number)
     WRAPPER_NO_CONTRACT
     _ASSERTE(number != NULL);
 
+    if (precision > NUMBER_MAXDIGITS)
+    {
+        precision = NUMBER_MAXDIGITS;
```

Comment:
Given that the format specifier is currently limited to `xx` (that is 0-99), maybe we should just set `NUMBER_MAXDIGITS` to `99`. Thoughts?

Reply:
Also, can users go over 99 digits with custom format specifiers? If so, we may need to tweak the algorithm to pad 0's past the "internal limit" (as per the IEEE requirement).

Reply:
I pushed a commit which changes `NUMBER_MAXDIGITS` to 99